ophyd/controls/area_detector.py

```diff
@@ -126,21 +127,10 @@ def _acquire_changed(self, value=None, old_value=None, **kwargs):
             self._run_subs(sub_type=self._SUB_ACQ_DONE)
             self._run_subs(sub_type=self._SUB_ACQ_CHECK)
 
-    def _stats_changed(self, **kwargs):
-        self._stats_counter += 1
-        self._run_subs(sub_type=self._SUB_ACQ_CHECK)
-
     def _check_if_finished(self, **kwargs):
-        if self._use_stats:
-            nstats = len(self._stats) * self._acq_num
-            if ((self._stats_counter == nstats) and
-                 (self._acq_count == self._acq_num)):
-                self._run_subs(sub_type=self._SUB_DONE)
-                self._reset_sub(self._SUB_DONE)
-        else:
-            if self._acq_count == self._acq_num:
-                self._run_subs(sub_type=self._SUB_DONE)
-                self._reset_sub(self._SUB_DONE)
+        if self._acq_count == self._acq_num:
```

Comment:
This still uses `self._acq_num`.  I still think that we should ensure that it is created in `__init__`.  Just because you _can_ dynamically add attributes to an object does not mean you _should_.